src/components/views/ViewDataTable/ViewDataTableSorting.tsx

```diff
@@ -0,0 +1,161 @@
+import { FormattedMessage } from 'react-intl';
+import { makeStyles } from '@material-ui/styles';
+import { useState } from 'react';
+import { Add, Delete, ImportExport } from '@material-ui/icons';
+import {
+    Badge,
+    Box,
+    Button,
+    Divider,
+    FormControl,
+    IconButton,
+    MenuItem,
+    Popover,
+    Select,
+    Typography,
+} from '@material-ui/core';
+import { GridColDef, GridSortModel } from '@mui/x-data-grid-pro';
+
+import ShiftKeyIcon from './ShiftKeyIcon';
+
+const useStyles = makeStyles({
+    deleteButton: {
+        padding: 6,
+    },
+    popover: {
+        borderRadius: 0,
+        minWidth: 450,
+        padding: 24,
+    },
+    shiftIcon: {
+        margin: '0 5px -17px 5px',
+    },
+    sortModelItem: {
+        padding: '0 0 8px 0',
+    },
+});
+
+interface ViewDataTableSortingProps {
+    gridColumns: GridColDef[];
+    setSortModel: (model: GridSortModel | []) => void;
+    sortModel: GridSortModel | [];
+}
+
+const ViewDataTableSorting: React.FunctionComponent<ViewDataTableSortingProps> = ({
+    gridColumns, setSortModel, sortModel,
+}) => {
+    const [anchorEl, setAnchorEl] = useState<HTMLButtonElement | null>(null);
+    const classes = useStyles();
+    const open = Boolean(anchorEl);
+    const id = open ? 'sort-options' : undefined;
+
+    const handleSortButtonClick = (event: React.SyntheticEvent<HTMLButtonElement>) => setAnchorEl(event.currentTarget);
+    const handlePopoverClose = () => setAnchorEl(null);
+
+    const handleChange = ({ name, value, field }: Record<string, string | undefined | unknown>) => {
+        const newSortModel = sortModel.map(item => {
+            if (name === 'delete' && item.field === field) return null;
+            else return item.field === field ?
+                { field: name === 'column' ? value : item.field, sort: name === 'direction' ? value : item.sort } : item;
```

Comment:
are the potential `name` values in an enum somewhere? possibly in the data grid library? if not, and if it's a finite number of known values, let's make an enum and use that in the type def for `handleChange`. Then we can compare against `FIELD_NAME.COLUMN` instead of strings.

Reply:
I have refactored this to use a simpler pattern for `handleChange()` which alleviates this.